### PR TITLE
Tests: adopt `AbsolutePath` in more of the PackageLoadingTests

### PR DESCRIPTION
**Pull request by compnerd**
This adjusts the Package Loading test suite to use `AbsolutePath` to
represent and perform path operations.  This allows flexibility for
differing path separator characters (e.g. `\` on Windows).

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Review by neonichu — APPROVED** [review]

File: Tests/PackageLoadingTests/TargetSourcesBuilderTests.swift [hunk 1]
```diff
@@ -62,13 +62,13 @@ class TargetSourcesBuilderTests: XCTestCase {
...
-            "/path/to/somefile.txt",
-            "/some/path.swift",
-            "/some/path/toBeCopied",
+            AbsolutePath("/Bar.swift").pathString,
+            AbsolutePath("/Foo.swift").pathString,
+            AbsolutePath("/Hello.something/hello.txt").pathString,
+            AbsolutePath("/file").pathString,
+            AbsolutePath("/path/to/somefile.txt").pathString,
+            AbsolutePath("/some/path.swift").pathString,
+            AbsolutePath("/some/path/toBeCopied").pathString,
```
Comment (tomerd) [inline Tests/PackageLoadingTests/TargetSourcesBuilderTests.swift:71]: remove `.pathString` from both ends of the comparison?

File: Tests/PackageLoadingTests/TargetSourcesBuilderTests.swift [hunk 2]
```diff
@@ -145,14 +145,16 @@ class TargetSourcesBuilderTests: XCTestCase {
         let contents = builder.computeContents().map{ $0.pathString }.sorted()
 
         XCTAssertEqual(contents, [
-            "/some.thing/hello.txt",
-            "/some/hello.swift",
+            AbsolutePath("/some.thing/hello.txt").pathString,
+            AbsolutePath("/some/hello.swift").pathString,
```
Comment (tomerd) [inline Tests/PackageLoadingTests/TargetSourcesBuilderTests.swift:149]: ditto

File: Tests/PackageLoadingTests/TargetSourcesBuilderTests.swift [hunk 3]
```diff
@@ -184,8 +186,8 @@ class TargetSourcesBuilderTests: XCTestCase {
         let contents = builder.computeContents().map{ $0.pathString }.sorted()
 
         XCTAssertEqual(contents, [
-            "/some.xcassets",
-            "/some/hello.swift"
+                root.appending(components: "some.xcassets").pathString,
+                root.appending(components: "some", "hello.swift").pathString,
```
Comment (tomerd) [inline Tests/PackageLoadingTests/TargetSourcesBuilderTests.swift:190]: ditto

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Pull request merged**